src/elementembeddings/_base.py

```
@@ -257,6 +258,40 @@ def calculate_umap(self, n_components: int = 2, standardise: bool = True, **kwar
+            init (str): Initialisation of the low-dimensional embedding, passed to
+                ``PaCMAP.fit_transform`` (e.g. ``"pca"`` or ``"random"``).
+            **kwargs: Other keyword arguments to be passed to PaCMAP.
+        """
+        if standardise:
+            if self.is_standardised:
+                embeddings_array = np.array(list(self.embeddings.values()))
+            else:
+                self.standardise(inplace=True)
+                embeddings_array = np.array(list(self.embeddings.values()))
+        else:
+            warnings.warn(
```

Ruff (0.15.17)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)